Sources/NIO/BaseSocket.swift

```diff
@@ -283,8 +285,8 @@ class BaseSocket: Selectable {
     /// - parameters:
     ///     - descriptor: The file descriptor to wrap.
     init(descriptor: CInt) {
+        precondition(descriptor >= 0, "invalid file descriptor")
```

Comment:
unclear to me whether this should be precondition or assert as I'm not sure if any user code could get here

Reply:
I think a precondition is fine here.